overlay_all_stats.py

```diff
@@ -57,27 +57,24 @@ def draw_power_rec_power(self):
             rec_power_text = "{0}".format(round(rec_power, 2))
         else:
             rec_power_text = "--"
-        self.data_canvas.draw_text(
-            rec_power_text, (340, self.text_height * 1)
-        )
+        self.data_canvas.draw_text(rec_power_text, (340, self.text_height * 1))
 
         # Display power
         if self.data["power"].is_valid():
             power_text = "{0}".format(round(power, 2))
-            if power > (rec_power + (rec_power * tolerance)):
-                power_colour = Colour.red
-            elif power > rec_power:
-                power_colour = Colour.green
+            if self.data["rec_power"].is_valid():
+                if power > (rec_power + (rec_power * tolerance)):
+                    power_colour = Colour.red
+                elif power > rec_power:
+                    power_colour = Colour.green
```

Comment:
This doesn't handle the case when `power` and `rec_power` are valid but `power <= rec_power`. When this occurs, a value is never assigned to `power_colour` and we get a crash.

To fix this, I would recommend just having `power_colour = Colour.black` by default above the if statements. That way, we don't need to worry about explicitly handling the "normal" cases where we colour the text black.